internal/controller/assets/deployment.go

```diff
@@ -741,3 +849,49 @@ func admissionDepUpdateStrategy(admission *falconv1alpha1.FalconAdmission) appsv
 		RollingUpdate: &rollingUpdateSettings,
 	}
 }
+
+func admissionDepWatcherEnvVars(admission *falconv1alpha1.FalconAdmission) []corev1.EnvVar {
+	envVars := []corev1.EnvVar{
+		corev1.EnvVar{
+			Name: "__CS_POD_NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "metadata.namespace",
+				},
+			},
+		},
+		corev1.EnvVar{
+			Name: "__CS_POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "metadata.name",
+				},
+			},
+		},
+		corev1.EnvVar{
+			Name: "__CS_POD_NODENAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "spec.nodeName",
+				},
+			},
+		},
+		corev1.EnvVar{
+			Name:  "__CS_SNAPSHOTS_ENABLED",
+			Value: admission.Spec.AdmissionConfig.GetSnapshotsEnabled(),
+		},
+		corev1.EnvVar{
+			Name:  "__CS_SNAPSHOT_INTERVAL",
+			Value: admission.Spec.AdmissionConfig.GetSnapshotsInterval(),
+		},
+		corev1.EnvVar{
+			Name:  "__CS_WATCH_EVENTS_ENABLED",
+			Value: admission.Spec.AdmissionConfig.GetWatcherEnabled(),
+		},
```

Comment:
The other side of the comment above, string-formatting of API properties is better done here.

Reply:
Updated